libs/SmartStore/SmartStoreTests/SFQuerySpecTests.m

```diff
@@ -0,0 +1,146 @@
+/*
+ Copyright (c) 2016, salesforce.com, inc. All rights reserved.
+ 
+ Redistribution and use of this software in source and binary forms, with or without modification,
+ are permitted provided that the following conditions are met:
+ * Redistributions of source code must retain the above copyright notice, this list of conditions
+ and the following disclaimer.
+ * Redistributions in binary form must reproduce the above copyright notice, this list of
+ conditions and the following disclaimer in the documentation and/or other materials provided
+ with the distribution.
+ * Neither the name of salesforce.com, inc. nor the names of its contributors may be used to
+ endorse or promote products derived from this software without specific prior written
+ permission of salesforce.com, inc.
+ 
+ THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR
+ IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
+ FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+ CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+ WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY
+ WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#import "SFQuerySpecTests.h"
+#import "SFQuerySpec.h"
+
+
+@implementation SFQuerySpecTests
+
+
+#pragma mark - setup and teardown
+
+- (void) setUp
+{
+    [super setUp];
+    [SFLogger setLogLevel:SFLogLevelDebug];
+}
+
+
+#pragma mark - tests
+- (void) testAllQuerySmartSql
```

Comment:
Since we introduced smart sql years ago, all query specs contain a smart sql field which is used for running the query. There is also a count smart sql field which is used when calling countQuery with the query spec. And with the remove by query story, we introduced a ids smart sql field used when doing a remove entries with the query spec.
All those have been indirectly tested in other places (by actually running queries), but now this test suite, looks at the actual smart sql calculated for various query spec types.